### ISSUE TITLE
Remove non-free Icons8 icons

### ISSUE THREAD

**Issue by Natureshadow** [iconify/collections-json#12]
The icons from Icons8 are mostly published under a non-free license (the "Good Boy License"). This license is meant to be funny, but makes it impossible to use the icons without legal issues in practice.

In general, it would be a good idea to only include icon sets licensed under an OSI-, FSF- or DFSG-approved license.

**Comment by cyberalien** [comment]
What exactly is the problem with using it? Just because license is not approved by various foundations, doesn't mean it is not valid. Functionally it is similar to MIT: no restrictions.

**Comment by Natureshadow** [comment]
https://icons8.com/good-boy-license

This is not entirely true. Arguably, this is what the authors **intended** to be true, but I am not even sure about that.

First off, many (most?) jurisdictions agree that a licensor has to grant all permissions explicitly that they want to grant. Simply waiving all rights is something that is not legally possible in most parts of the world (the fact that the copyright law from the actual place of residence of an author at the very moment they created the work is the basis for this does not make anything easier).

The license starts of with the preamble: "Please do whatever your mom would approve of." As you probably agree, this is a requirement that cannot objectively be fulfilled (the concept of objectivity is a very basic rule in many legal questions). We can safely assume that this preamble can thus not be part of the license, either making it void entirely, or, if we rely on an implicit escape clause, the rest of the license remains valid. (Or, if this sentence is a valid license clause: I have talked to young people who's parents did explicitly not approve of them learning how to create web apps; such parental views seem to be quite common in e.g. India — does it mean the licensor explicitly turns that into a legal requirement to listen to these parents?)

The rest of the license is an explicit list of things that are allowed or not allowed. On the allow list, this is strictly limited to:

* Download — ok
* Change — ok
* Fork on GitHub — ok, but not on any other Git hosting platform, or locally, or anywhere else than GitHub at all?

It does not permit redistribution of modified versions, it does not even grant usage of the icons after download, or anything else.

On the deny list, we have:

* No tattoos — What does this mean? Is it forbidden to use the icons as an actual tattoo on a person's skin? Is use of the icons forbidden for people who have a tattoo?
* No touching (with unwashed hands) — Does that include touching a button using the icon on a smartphone display?
* No exchanging for drugs — That's somewhat along the licence of the infamous "No Evil Licence", somehow, or whatever

All of these seem to be a joke about that "what your mom would approve of".

While we can probably assume that the original authors of Icons8 are not serious about these terms, a license is still a legal document. Licenses are annoying, but as it stands, are a requirement for most of the Earth's population to be able to use a work legally. And I have, personally, fought against licensing trolls who abuse unclear wording in, even professional and serious, Open Source licenses to directly harm users — a license like the Good Boy License is a perfect match for such people.

All of this is why there are organisations which certify licenses for their fitness to be used in open source products, and why many companies still fear the free software ecosystem because of somewhat unclear licensing situations.

Please do not help makign all this more difficult than it should be. Instead, if you really want to include the icon sets from Icons8, please ask them to dual-license under MIT, as they already did for one of their icon sets (https://github.com/icons8/flat-color-icons/issues/5, where they implicitly agreed with the view that the Good Boy License alone is difficult).

**Comment by cyberalien** [comment]
That makes sense. But why not contact Icons8 with your concerns to ask them to change license or add second license option?

**Comment by Natureshadow** [comment]
> That makes sense. But why not contact Icons8 with your concerns to ask them to change license or add second license option?

I am the author of [django-iconify](https://pypi.org/project/django-iconify/), an Iconify API server written in Python and intended for use in Django applications (by the way, what would be the right place for opening a merge request to get this hosting option listed on iconify.design ☺?).

Therefore, I am not so much interested in the licensing of a specific icon set, but my primary concern is the distribution. A distribution has to consider the consequences of shipping the parts it distributes, plus the consequences of shipping them as a collection, and I think it is the distributor's obligation to ensure their distribution is legal and does not help cause harm to their users. In that regard, Iconify makes a promise on its front page:

> This license does not apply to icons. Icons are released under different licenses, see each collection for details. Icon collections available by default are all licensed under some kind of open source or free license, making it possible to use them in Iconify project.

While also hinting users at their own obligation to review specific licenses, it also suggests that the situation might be somewhat safe.

So, in conclusion, I consider it our shared responsibility (you as distributor of the default data set, and me as the provider making it available through an API hosting option) to help users in that regard.

As for djano-iconify, I plan to add a configuration parameter to:

* Selectively enable and disable icon sets
* Selectively enable and disable licenses
* Enable only icon sets who's SPDX license identifier is a license approved by an Open Source standards body

But that is out of scope here.

**Comment by cyberalien** [comment]
Nice! Thank you for making that API server and this thread is a great place to mention it. Wasn't aware of it before your post. I'll add it to documentation.

As for icon sets, I'll contact Icons8 to see if they are willing to change license.

**Comment by cyberalien** [comment]
So good news. I've contacted Icons8, they confirmed that it is double licensed: their custom license and MIT.

License in json files will change in next icon sets update tomorrow.

**Comment by cyberalien** [comment]
License for icon sets changed and django-iconify added to documentation.

Thanks for using Iconify! Please create more issues if you have any questions or concerns.
